java-frontend/src/main/java/org/sonar/java/se/ExplodedGraphWalker.java

```diff
@@ -480,27 +494,62 @@ private void executeMethodInvocation(MethodInvocationTree mit) {
     logState(mit);
 
+    programState = unstack.state;
+    // Enqueue exceptional paths
+    node.programPoint.block.exceptions().forEach(b -> enqueue(new ExplodedGraph.ProgramPoint(b, 0), programState.clearStack(), !b.isCatchBlock()));
+
     // get method behavior for method with known declaration (ie: within the same file)
-    Tree declaration = mit.symbol().declaration();
+    MethodBehavior methodInvokedBehavior = null;
+    Symbol methodSymbol = mit.symbol();
+    Tree declaration = methodSymbol.declaration();
     if(declaration != null) {
-      // (FIXME) nothing done of the invoked behavior
-      MethodBehavior methodInvokedBehavior = symbolicExecutionVisitor.execute((MethodTree) declaration);
+      methodInvokedBehavior = symbolicExecutionVisitor.execute((MethodTree) declaration);
     }
-
-
-
-    // Enqueue exceptional paths
-    node.programPoint.block.exceptions().forEach(b -> enqueue(new ExplodedGraph.ProgramPoint(b, 0), programState, !b.isCatchBlock()));
     final SymbolicValue resultValue = constraintManager.createMethodSymbolicValue(mit, unstack.values);
-    programState = programState.stackValue(resultValue);
-    if (isNonNullMethod(mit.symbol())) {
-      programState = programState.addConstraint(resultValue, ObjectConstraint.NOT_NULL);
-    } else if (OBJECT_WAIT_MATCHER.matches(mit)) {
-      programState = programState.resetFieldValues(constraintManager);
+    if (methodInvokedBehavior != null && methodInvokedBehavior.isComplete() && methodCanNotBeOverriden(methodSymbol)) {
+      List<SymbolicValue> invocationArguments = invocationArguments(unstack.values);
+      methodInvokedBehavior.yields()
+        .stream()
+        .filter(yield -> !yield.exception)
+        .flatMap(yield -> yield.statesAfterInvocation(invocationArguments, programState, () -> resultValue).stream())
+        .forEach(psYield -> {
+          ProgramState ps = psYield;
+          if (isNonNullMethod(methodSymbol)) {
+            ps = ps.addConstraint(ps.peekValue(), ObjectConstraint.NOT_NULL);
+          } else if (OBJECT_WAIT_MATCHER.matches(mit)) {
+            ps = ps.resetFieldValues(constraintManager);
+          }
+          checkerDispatcher.syntaxNode = mit;
+          checkerDispatcher.addTransition(ps);
+          clearStack(mit);
+        });
+    } else {
+      programState = programState.stackValue(resultValue);
+      if (isNonNullMethod(methodSymbol)) {
+        programState = programState.addConstraint(resultValue, ObjectConstraint.NOT_NULL);
+      } else if (OBJECT_WAIT_MATCHER.matches(mit)) {
+        programState = programState.resetFieldValues(constraintManager);
+      }
+      checkerDispatcher.executeCheckPostStatement(mit);
+      clearStack(mit);
+    }
+  }
+
+  private static boolean methodCanNotBeOverriden(Symbol methodSymbol) {
+    if ((((JavaSymbol.MethodJavaSymbol) methodSymbol).flags() & Flags.NATIVE) != 0) {
+      return false;
     }
+    return !methodSymbol.isAbstract() &&
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1067)